src/go/k8s/pkg/resources/certmanager/ca_certificate_bundle_test.go

```diff
@@ -0,0 +1,267 @@
+
+	fake "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var fakeK8sClient = fake.NewClientBuilder().Build()
+
+func init() {
+	v1alpha1.AddToScheme(fakeK8sClient.Scheme())
+}
+
+func TestCreateCACertBundle(t *testing.T) {
+	nsPrefix := "test-ns"
+	bundleCACertSecretSuffix := "ca-bundle"
+	clusterName := "test-cluster-create"
+
+	pandaCluster := &v1alpha1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterName,
+		},
+		Spec: v1alpha1.ClusterSpec{
+			Replicas:      pointer.Int32(1),
+			Configuration: v1alpha1.RedpandaConfig{},
+		},
+	}
+
+	caCert, err := getCertSecretFromFile("./testdata/ca-cert-secret.yaml")
+	assert.NoError(t, err)
+	caCert.SetNamespace(nsPrefix)
+	err = fakeK8sClient.Create(context.TODO(), caCert)
+	assert.NoError(t, err)
+
+	tlsCert, err := getCertSecretFromFile("./testdata/tls-cert-secret.yaml")
+	assert.NoError(t, err)
+	tlsCert.SetNamespace(nsPrefix)
+	err = fakeK8sClient.Create(context.TODO(), tlsCert)
+	assert.NoError(t, err)
+
+	expectedOneCACertData := getExpectedCACert("./testdata/expected-one-ca-crt.pem")
+	expectedTwoCACertData := getExpectedCACert("./testdata/expected-two-ca-crt.pem")
+	expectedThreeCACertData := getExpectedCACert("./testdata/expected-three-ca-crt.pem")
+
+	table := []struct {
+		scenario         string
+		certs            []*types.NamespacedName
+		expectedBundleCA []byte
+	}{
+		{
+			scenario:         "Get CA cert from secret having ca.crt",
+			certs:            []*types.NamespacedName{{Name: caCert.Name, Namespace: caCert.Namespace}},
+			expectedBundleCA: expectedOneCACertData,
+		},
+		{
+			scenario:         "Get CA certs from secret having tls.crt",
+			certs:            []*types.NamespacedName{{Name: tlsCert.Name, Namespace: tlsCert.Namespace}},
+			expectedBundleCA: expectedTwoCACertData,
+		},
+		{
+			scenario: "Get CA certs from secret having ca.crt and secret having tls.crt",
+			certs: []*types.NamespacedName{
+				{Name: caCert.Name, Namespace: caCert.Namespace},
+				{Name: tlsCert.Name, Namespace: tlsCert.Namespace},
+			},
+			expectedBundleCA: expectedThreeCACertData,
+		},
+	}
+	for i, tt := range table {
+		t.Run(fmt.Sprintf("case-%d: %s", i, tt.scenario), func(t *testing.T) {
+			ns := fmt.Sprintf("%s-%d", nsPrefix, i)
+			pandaCluster.SetNamespace(ns)
+
+			caCertBundle := certmanager.NewCACertificateBundle(fakeK8sClient, fakeK8sClient.Scheme(), pandaCluster,
+				tt.certs, bundleCACertSecretSuffix, logr.Discard())
+
+			err := caCertBundle.Ensure(context.TODO())
+			assert.NoError(t, err)
+
+			var bundledCASecret corev1.Secret
+			err = fakeK8sClient.Get(context.TODO(), types.NamespacedName{Name: fmt.Sprintf("%s-%s", clusterName, bundleCACertSecretSuffix), Namespace: ns}, &bundledCASecret)
+			assert.NoError(t, err)
+			assert.Equal(t, tt.expectedBundleCA, bundledCASecret.Data["ca.crt"])
+		})
+	}
+
+	err = fakeK8sClient.Delete(context.TODO(), caCert)
+	assert.NoError(t, err)
+	err = fakeK8sClient.Delete(context.TODO(), tlsCert)
+	assert.NoError(t, err)
```

Comment:
Can you build fakeK8sClient in every test? This teardown can be then removed. Creating fake client should not be consider as heavy long running operation.

Reply:
Good idea. Done in the next commit.